build.sbt

```diff
@@ -7,55 +7,61 @@ import org.scalajs.sbtplugin.cross.CrossProject
 
 lazy val checkHeaders = taskKey[Unit]("Fail the build if createHeaders is not up-to-date")
 
-val monocleVersion = "1.2.1"
-val scalazVersion = "7.2.1"
-// Latest version built against scalacheck 1.12.5
-// doesn't support scala.js yet, until then tests are JVM-only
-val specs2Version = "3.7"
+// (monocle) doesn't support scala.js yet, until then tests are JVM-only
+val scalazVersion     = "7.2.6"
+val specs2Version     = "3.8.5.1"
+val scalacheckVersion = "1.13.3"
+val disciplineVersion = "0.7.1"
 
-val testDependencies = libraryDependencies ++= Seq(
-  "org.typelevel"  %% "discipline"                % "0.4"          % "test",
-  "com.github.julien-truffaut" %% "monocle-law"   % monocleVersion % "test",
-  "org.scalaz"     %% "scalaz-scalacheck-binding" % scalazVersion  % "test",
-  "org.typelevel"  %% "scalaz-specs2"             % "0.4.0"        % "test",
-  "org.specs2"     %% "specs2-core"               % specs2Version  % "test" force(),
-  "org.specs2"     %% "specs2-scalacheck"         % specs2Version  % "test" force(),
-  // `scalaz-scalack-binding` is built with `scalacheck` 1.12.5 so we are stuck
-  // with that version
-  "org.scalacheck" %% "scalacheck"                % "1.12.5"       % "test" force()
-)
+def monocleVersion(sv: String) = sv match {
+  case "2.11" => "1.3.0"
+  case _      => "1.4.0-SNAPSHOT"
+}
+def versionDeps(sv: String): Seq[ModuleID] = sv match {
+  case "2.11" => Seq(compilerPlugin("com.milessabin" % "si2712fix-plugin" % "1.2.0" cross CrossVersion.full))
```

Comment:
What about using the typelevel compiler for 2.11? Too many differences to make it easy to cross-build?